src/Components/Footnote.tsx

```diff
@@ -1,15 +1,22 @@
-import PropTypes from 'prop-types';
+import type { ComponentChildren } from 'preact';
 import { Text, IntlProvider } from 'preact-i18n';
 import { useEffect, useRef } from 'preact/hooks';
 
-const Footnote = (props) => {
+type FootnoteProps = {
+    index: number;
+    id: string;
+    children: ComponentChildren;
+};
+
+const Footnote = (props: FootnoteProps) => {
     const detailsRef = useRef(null);
 
     useEffect(() => {
-        //Adapted after: https://usehooks.com/useOnClickOutside/
-        const listener = (event) => {
-            if (detailsRef.current?.open && !detailsRef.current?.contains(event.target)) {
-                detailsRef.current.removeAttribute('open');
+        // Adapted after: https://usehooks.com/useOnClickOutside/
+        const listener = (event: MouseEvent | TouchEvent) => {
+            const details_el = detailsRef.current as HTMLDetailsElement | null;
+            if (details_el?.open && !details_el?.contains(event.target as Element)) {
```

Comment:
No, that would break it. This is one of the few cases where we explicitly need whatever element caused the event, not the one that has the event handler.